mlflow/store/artifact/dbfs_artifact_repo.py

```diff
@@ -159,11 +162,23 @@ def dbfs_artifact_repo_factory(artifact_uri):
 
     This factory method is used with URIs of the form ``dbfs:/<path>``. DBFS-backed artifact
     storage can only be used together with the RestStore.
+
```

Comment:
@arjundc-db Can we add a test for this method against various types of URIs to ensure that the correct repos are returned? This will help ensure that we're not accidentally returning ACL'ed `DatabricksArtifactRepo` instances for non-ACL'ed DBFS URIs.

Reply:
@dbczumar I added a single test here : https://github.com/mlflow/mlflow/pull/2911/files#diff-f78e8f823247e329ce4e4d004bd40f7cR44-R47
I could add more tests if needed, however since it only depends on the prefix `databricks/mlflow-tracking/`, adding additional tests may be redundant. 
Note that the use of `extract_and_normalize_path` is not used here/

Reply:
Looks great - sorry for missing this!